src/appshell/internal/applicationuiactions.cpp

```diff
@@ -252,6 +258,10 @@ bool ApplicationUiActions::actionEnabled(const UiAction& act) const
 
 bool ApplicationUiActions::actionChecked(const UiAction& act) const
 {
+    if (act.code == FULL_SCREEN_CODE) {
+        return mainWindow() ? mainWindow()->isFullScreen() : false;
```

Comment:
Now that `mainWindow()` exists _always_, you don't need the nullptr check anymore

Reply:
Done with all the changes, you can check